packages/neuron-wallet/src/services/node.ts

```diff
@@ -187,6 +200,56 @@ class NodeService {
         return false
       })
   }
+
+  private getInternalNodeVersion() {
+    const appPath = electronApp.isPackaged ? electronApp.getAppPath() : path.join(__dirname, '../../../..')
+    const ckbVersionPath = path.join(appPath, '.ckb-version')
+    if (fs.existsSync(ckbVersionPath)) {
+      try {
+        return fs
+          .readFileSync(ckbVersionPath, 'utf8')
+          ?.split('\n')?.[0]
+          ?.slice(1)
+      } catch (err) {
+        logger.error('App\t: get ckb node version failed')
+      }
+    }
+  }
+
+  private async verifyNodeVersion() {
+    const network = NetworksService.getInstance().getCurrent()
+    const localNodeInfo = await new RpcService(network.remote).getLocalNodeInfo()
+    const internalNodeVersion = this.getInternalNodeVersion()
+    const [internalMajor, internalMinor] = internalNodeVersion?.split('.') ?? []
+    const [externalMajor, externalMinor] = localNodeInfo.version?.split('.') ?? []
+
+    if (internalMajor !== externalMajor || internalMinor !== externalMinor) {
```

Comment:
Good enough for now, but once `major > 0`, a higher `minor` should be acceptable because it won't change breakingly.